lib/config/config.go

```diff
@@ -225,6 +244,12 @@ found:
 		return err
 	}
 
+	// Add a default GUI configuration if no one has been given.
+	if len(cfg.RawGUIs) == 0 {
+		cfg.RawGUIs = append(cfg.RawGUIs, GUIConfiguration{})
+		util.SetDefaults(&cfg.RawGUIs[0])
+	}
```

Comment:
This is a bit wishy washy. Create, add to slice, then index access back from the slice.

Maybe this:

```
         // Add a default GUI configuration if no one has been given.
        if len(cfg.RawGUIs) == 0 {
                var guiCfg GUIConfiguration
                util.SetDefaults(&guiCfg)
		cfg.RawGUIs = []GUIConfiguration{guiCfg}
        }
```